src/jit/lsraxarch.cpp

```diff
@@ -2606,38 +2606,16 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
                     srcCount += BuildOperandUses(op2);
                     srcCount += BuildDelayFreeUses(op3);
                 }
-                else if (op1->isContained())
+                else
                 {
-                    // 231 form: op3 = (op2 * op3) + [op1]
+                    // 231 form: op1 = (op2 * op3) + [op1]
 
                     tgtPrefUse = BuildUse(op3);
 
                     srcCount += BuildOperandUses(op1);
                     srcCount += BuildDelayFreeUses(op2);
                     srcCount += 1;
                 }
-                else
-                {
-                    // 213 form: op1 = (op2 * op1) + op3
-
-                    if (copiesUpperBits)
```

Comment:
Similar comment, pretty sure we need to preserve this logic since op1 impacts determinism.

Reply:
For clarity as well, I think it makes sense to continue to keep the reg-only case separate, even if it's using the same form as the case above.